---
title: "Chordata Whole genomes data of nucleus in comparison with mitochondria"
author: "Alina G. Mikhailova"
date: "2023-11-2"
output:    
   html_document:
      toc: true
      toc_depth: 3
      toc_float: true
      number_section: true
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
```

```{r echo=FALSE, message=FALSE}
rm(list=ls(all=TRUE))
if (!require(caper)) install.packages("caper")
if (!require(geiger)) install.packages("geiger")
if (!require(ggpubr)) install.packages("ggpubr")
if (!require(dplyr)) install.packages("dplyr")
if (!require(tidyr)) install.packages("tidyr")
library(tidyr)
library(caper)
library(geiger)
library(ggpubr)
library(dplyr)
```

# Reading whole genomes database and obtaining neutral nucleotide fractions in nucleus

```{r}
data = read.csv("../data/1raw/Nuclear_codon_usage.csv")
data$Species = gsub(" ", "_", data$Species)
data$NeutralA = as.numeric(data$GCA..Ala.) + as.numeric(data$CGA..Arg.) + as.numeric(data$GGA..Gly.) + as.numeric(data$CTA..Leu.)  + as.numeric(data$CTA..Leu.)  + as.numeric(data$CCA..Pro.)  + as.numeric(data$TCA..Ser.)  + as.numeric(data$ACA..Thr.) + as.numeric(data$GTA..Val.)
data$NeutralT = as.numeric(data$GCT..Ala.) + as.numeric(data$CGT..Arg.) + as.numeric(data$GGT..Gly.) + as.numeric(data$CTT..Leu.)  + as.numeric(data$CTT..Leu.)  + as.numeric(data$CCT..Pro.)  + as.numeric(data$TCT..Ser.)  + as.numeric(data$ACT..Thr.) + as.numeric(data$GTT..Val.)
data$NeutralG = as.numeric(data$GCG..Ala.) + as.numeric(data$CGG..Arg.) + as.numeric(data$GGG..Gly.) + as.numeric(data$CTG..Leu.)  + as.numeric(data$CTG..Leu.)  + as.numeric(data$CCG..Pro.)  + as.numeric(data$TCG..Ser.)  + as.numeric(data$ACG..Thr.) + as.numeric(data$GTG..Val.)
data$NeutralC = as.numeric(data$GCC..Ala.) + as.numeric(data$CGC..Arg.) + as.numeric(data$GGC..Gly.) + as.numeric(data$CTC..Leu.)  + as.numeric(data$CTC..Leu.)  + as.numeric(data$CCC..Pro.)  + as.numeric(data$TCC..Ser.)  + as.numeric(data$ACC..Thr.) + as.numeric(data$GTC..Val.)

sinnuccontentn = aggregate(list(data$NeutralA,data$NeutralT,data$NeutralG,data$NeutralC), by = list(data$Species, data$Class), FUN = sum)
names(sinnuccontentn) = c('Species', 'Class', 'NeutralA','NeutralT','NeutralG','NeutralC')
sinnuccontentn$FrA = sinnuccontentn$NeutralA / (sinnuccontentn$NeutralA + sinnuccontentn$NeutralT + sinnuccontentn$NeutralG + sinnuccontentn$NeutralC)
sinnuccontentn$FrT = sinnuccontentn$NeutralT / (sinnuccontentn$NeutralA + sinnuccontentn$NeutralT + sinnuccontentn$NeutralG + sinnuccontentn$NeutralC) 
sinnuccontentn$FrG = sinnuccontentn$NeutralG / (sinnuccontentn$NeutralA + sinnuccontentn$NeutralT + sinnuccontentn$NeutralG + sinnuccontentn$NeutralC) 
sinnuccontentn$FrC = sinnuccontentn$NeutralC / (sinnuccontentn$NeutralA + sinnuccontentn$NeutralT + sinnuccontentn$NeutralG + sinnuccontentn$NeutralC) 
sinnuccontentn$TGskew = (sinnuccontentn$FrT - sinnuccontentn$FrG)+(sinnuccontentn$FrT + sinnuccontentn$FrG)
sinnuccontentn$ACskew = (sinnuccontentn$FrA - sinnuccontentn$FrC)+(sinnuccontentn$FrA + sinnuccontentn$FrC)
#cor.test(sinnuccontentn$TGskew, sinnuccontentn$ACskew)
sinnuccontentn = sinnuccontentn %>% rowwise() %>% mutate(meanTGACskew = median(c(TGskew, ACskew)))

table(sinnuccontentn$Class)

```
# Reading whole genomes database and obtaining neutral nucleotide fractions in mitochondria

```{r}
data = read.csv("../data/1raw/Mitochondrial_codon_usage.csv")
data$Species = gsub(" ", "_", data$Species)
data$NeutralA = as.numeric(data$GCA..Ala.) + as.numeric(data$CGA..Arg.) + as.numeric(data$GGA..Gly.) + as.numeric(data$CTA..Leu.)  + as.numeric(data$CTA..Leu.)  + as.numeric(data$CCA..Pro.)  + as.numeric(data$TCA..Ser.)  + as.numeric(data$ACA..Thr.) + as.numeric(data$GTA..Val.)
data$NeutralT = as.numeric(data$GCT..Ala.) + as.numeric(data$CGT..Arg.) + as.numeric(data$GGT..Gly.) + as.numeric(data$CTT..Leu.)  + as.numeric(data$CTT..Leu.)  + as.numeric(data$CCT..Pro.)  + as.numeric(data$TCT..Ser.)  + as.numeric(data$ACT..Thr.) + as.numeric(data$GTT..Val.)
data$NeutralG = as.numeric(data$GCG..Ala.) + as.numeric(data$CGG..Arg.) + as.numeric(data$GGG..Gly.) + as.numeric(data$CTG..Leu.)  + as.numeric(data$CTG..Leu.)  + as.numeric(data$CCG..Pro.)  + as.numeric(data$TCG..Ser.)  + as.numeric(data$ACG..Thr.) + as.numeric(data$GTG..Val.)
data$NeutralC = as.numeric(data$GCC..Ala.) + as.numeric(data$CGC..Arg.) + as.numeric(data$GGC..Gly.) + as.numeric(data$CTC..Leu.)  + as.numeric(data$CTC..Leu.)  + as.numeric(data$CCC..Pro.)  + as.numeric(data$TCC..Ser.)  + as.numeric(data$ACC..Thr.) + as.numeric(data$GTC..Val.)

sinnuccontentm = aggregate(list(data$NeutralA,data$NeutralT,data$NeutralG,data$NeutralC), by = list(data$Species, data$Class), FUN = sum)
names(sinnuccontentm) = c('Species', 'Class', 'NeutralA','NeutralT','NeutralG','NeutralC')
sinnuccontentm$FrA = sinnuccontentm$NeutralA / (sinnuccontentm$NeutralA + sinnuccontentm$NeutralT + sinnuccontentm$NeutralG + sinnuccontentm$NeutralC)
sinnuccontentm$FrT = sinnuccontentm$NeutralT / (sinnuccontentm$NeutralA + sinnuccontentm$NeutralT + sinnuccontentm$NeutralG + sinnuccontentm$NeutralC) 
sinnuccontentm$FrG = sinnuccontentm$NeutralG / (sinnuccontentm$NeutralA + sinnuccontentm$NeutralT + sinnuccontentm$NeutralG + sinnuccontentm$NeutralC) 
sinnuccontentm$FrC = sinnuccontentm$NeutralC / (sinnuccontentm$NeutralA + sinnuccontentm$NeutralT + sinnuccontentm$NeutralG + sinnuccontentm$NeutralC) 
sinnuccontentm$GAskew.heavy = (sinnuccontentm$FrC-sinnuccontentm$FrT)/(sinnuccontentm$FrT+sinnuccontentm$FrC)

table(sinnuccontentm$Class)
```

# Reading whole genomes database and obtaining neutral nucleotide fractions in mitochondria (our dataset)
```{r}
sinnuccontentmOUR = read.table("../data/2derived/AllGenesCodonUsageNoOverlap.txt", header = TRUE, sep = '\t')
sinnuccontentmOUR = aggregate(list(sinnuccontentmOUR$NeutralA,sinnuccontentmOUR$NeutralT,sinnuccontentmOUR$NeutralG,sinnuccontentmOUR$NeutralC), by = list(sinnuccontentmOUR$Species), FUN = sum)
names(sinnuccontentmOUR) = c('Species','NeutralA','NeutralT','NeutralG','NeutralC')
sinnuccontentmOUR$FrA = sinnuccontentmOUR$NeutralA / (sinnuccontentmOUR$NeutralA + sinnuccontentmOUR$NeutralT + sinnuccontentmOUR$NeutralG + sinnuccontentmOUR$NeutralC)
sinnuccontentmOUR$FrT= sinnuccontentmOUR$NeutralT / (sinnuccontentmOUR$NeutralA + sinnuccontentmOUR$NeutralT + sinnuccontentmOUR$NeutralG + sinnuccontentmOUR$NeutralC) 
sinnuccontentmOUR$FrG = sinnuccontentmOUR$NeutralG / (sinnuccontentmOUR$NeutralA + sinnuccontentmOUR$NeutralT + sinnuccontentmOUR$NeutralG + sinnuccontentmOUR$NeutralC) 
sinnuccontentmOUR$FrC = sinnuccontentmOUR$NeutralC / (sinnuccontentmOUR$NeutralA + sinnuccontentmOUR$NeutralT + sinnuccontentmOUR$NeutralG + sinnuccontentmOUR$NeutralC) 
sinnuccontentmOUR$GAskew.heavy = (sinnuccontentmOUR$FrC-sinnuccontentmOUR$FrT)/(sinnuccontentmOUR$FrT+sinnuccontentmOUR$FrC)

table(sinnuccontentmOUR$Class)
###Adding taxa
taxa = read.table("../data/2derived/TaxaFromKostya.txt", header = T)
AA = read.table("../data/1raw/anage_data.txt", header = TRUE, sep = '\t')
AA$Species = paste(AA$Genus,AA$Species,sep = '_')
AAtaxa = AA[c("Species", "Class")]
taxa = rbind(taxa, AAtaxa)
taxa = taxa[!duplicated(taxa$Species),]
sinnuccontentmOUR = merge(sinnuccontentmOUR, taxa, all.x = T)
#temp = merge(sinnuccontentm,sinnuccontentmOUR, by="Species")
#cor.test(temp$GAskew.heavy.x, temp$GAskew.heavy.y)
```
# Merging both tables of fractions and correlating skews
```{r message=FALSE}
merged_fr = merge(sinnuccontentm,sinnuccontentn, by="Species")
cor.test(merged_fr$GAskew.heavy, merged_fr$meanTGACskew, method = "spearman")
```

```{r message=FALSE}

```

# Figures
## Figures for nuclear codon usage
```{r}
ggviolin(sinnuccontentn, x = "Class", y = "FrT", select = c("Actinopterygii", "Amphibia", "Reptilia", "Mammalia","Aves"), ylab = "FrT",
         order=c("Actinopterygii", "Amphibia", "Reptilia", "Mammalia","Aves"), add = "boxplot", fill="Class", palette=c("#6760db", "#7849bf", "#9145c4", "#c73a69", "#c2464c"), yscale = "log2")
ggviolin(sinnuccontentn, x = "Class", y = "FrG", select = c("Actinopterygii", "Amphibia", "Reptilia", "Mammalia","Aves"), ylab = "FrG",
         order=c("Actinopterygii", "Amphibia", "Reptilia", "Mammalia","Aves"), add = "boxplot", fill="Class", palette=c("#6760db", "#7849bf", "#9145c4", "#c73a69", "#c2464c"), yscale = "log2")
ggviolin(sinnuccontentn, x = "Class", y = "meanTGACskew", select = c("Actinopterygii", "Amphibia", "Reptilia", "Mammalia","Aves"), ylab = "meanTGACskew",
         order=c("Actinopterygii", "Amphibia", "Reptilia", "Mammalia","Aves"), add = "boxplot", fill="Class", palette=c("#6760db", "#7849bf", "#9145c4", "#c73a69", "#c2464c"))

colder <- c("Actinopterygii", "Amphibia", "Reptilia")
warmer <- c("Mammalia","Aves")
wilcox.test(sinnuccontentn[sinnuccontentn$Class %in% colder,]$meanTGACskew, sinnuccontentn[sinnuccontentn$Class %in% warmer,]$meanTGACskew, paired=F) 

ggviolin(sinnuccontentn, x = "Class", y = "FrT", select = c("Mammalia","Aves"), ylab = "FrT",
         order=c("Mammalia","Aves"), add = "boxplot", fill="Class", palette=c("#c73a69", "#c2464c"), yscale = "log2")
wilcox.test(sinnuccontentn[sinnuccontentn$Class == "Mammalia",]$FrT, sinnuccontentn[sinnuccontentn$Class == "Aves",]$FrT, paired=F) 
ggviolin(sinnuccontentn, x = "Class", y = "FrT", select = c("Actinopterygii", "Reptilia"), ylab = "FrT",
         order=c("Actinopterygii", "Reptilia"), add = "boxplot", fill="Class", palette=c("#6760db", "#9145c4"), yscale = "log2")
wilcox.test(sinnuccontentn[sinnuccontentn$Class == "Actinopterygii",]$FrT, sinnuccontentn[sinnuccontentn$Class == "Reptilia",]$FrT, paired=F) 


```
## Figures for mitochondrial codon usage
```{r}
ggviolin(sinnuccontentm, x = "Class", y = "FrT", select = c("Actinopterygii", "Amphibia", "Reptilia", "Mammalia","Aves"), ylab = "FrA",
         order=c("Actinopterygii", "Amphibia", "Reptilia", "Mammalia","Aves"), add = "boxplot", fill="Class", palette=c("#6760db", "#7849bf", "#9145c4", "#c73a69", "#c2464c"), yscale = "log2")
ggviolin(sinnuccontentm, x = "Class", y = "FrC", select = c("Actinopterygii", "Amphibia", "Reptilia", "Mammalia","Aves"), ylab = "FrG",
         order=c("Actinopterygii", "Amphibia", "Reptilia", "Mammalia","Aves"), add = "boxplot", fill="Class", palette=c("#6760db", "#7849bf", "#9145c4", "#c73a69", "#c2464c"), yscale = "log2")
ggviolin(sinnuccontentm, x = "Class", y = "GAskew.heavy", select = c("Actinopterygii", "Amphibia", "Reptilia", "Mammalia","Aves"), ylab = "GAskew",
         order=c("Actinopterygii", "Amphibia", "Reptilia", "Mammalia","Aves"), add = "boxplot", fill="Class", palette=c("#6760db", "#7849bf", "#9145c4", "#c73a69", "#c2464c"))

colder <- c("Actinopterygii", "Amphibia", "Reptilia")
warmer <- c("Mammalia","Aves")
wilcox.test(sinnuccontentm[sinnuccontentm$Class %in% colder,]$GAskew.heavy, sinnuccontentm[sinnuccontentm$Class %in% warmer,]$GAskew.heavy, paired=F) 

```

## Figures for mitochondrial codon usage (our dataset)
```{r}
ggviolin(sinnuccontentmOUR, x = "Class", y = "FrT", select = c("Actinopterygii", "Amphibia", "Reptilia", "Mammalia","Aves"), ylab = "FrA",
         order=c("Actinopterygii", "Amphibia", "Reptilia", "Mammalia","Aves"), add = "boxplot", fill="Class", palette=c("#6760db", "#7849bf", "#9145c4", "#c73a69", "#c2464c"), yscale = "log2")
ggviolin(sinnuccontentmOUR, x = "Class", y = "FrC", select = c("Actinopterygii", "Amphibia", "Reptilia", "Mammalia","Aves"), ylab = "FrG",
         order=c("Actinopterygii", "Amphibia", "Reptilia", "Mammalia","Aves"), add = "boxplot", fill="Class", palette=c("#6760db", "#7849bf", "#9145c4", "#c73a69", "#c2464c"), yscale = "log2")
ggviolin(sinnuccontentmOUR, x = "Class", y = "GAskew.heavy", select = c("Actinopterygii", "Amphibia", "Reptilia", "Mammalia","Aves"), ylab = "GAskew",
         order=c("Actinopterygii", "Amphibia", "Reptilia", "Mammalia","Aves"), add = "boxplot", fill="Class", palette=c("#6760db", "#7849bf", "#9145c4", "#c73a69", "#c2464c"))

colder <- c("Actinopterygii", "Amphibia", "Reptilia")
warmer <- c("Mammalia","Aves")
wilcox.test(sinnuccontentmOUR[sinnuccontentmOUR$Class %in% colder,]$GAskew.heavy, sinnuccontentmOUR[sinnuccontentmOUR$Class %in% warmer,]$GAskew.heavy, paired=F) 

```


# Exploring Amphybians
```{r}
data = read.csv("../data/1raw/Nuclear_codon_usage.csv"); data$Species = gsub(" ", "_", data$Species) #data[data$Class == "Amphibia",]

colnames(data)
data = data %>% mutate(sum = rowSums(across(where(is.numeric)), na.rm=TRUE))
data = data %>% mutate(sum_pro = rowSums(across(c("CCA..Pro.", "CCC..Pro.", "CCG..Pro.", "CCT..Pro."))))
data = data %>% mutate(sum_phe = rowSums(across(c("TTT..Phe.", "TTC..Phe."))))
data = data %>% mutate(sum_lys = rowSums(across(c("AAA..Lys.", "AAG..Lys."))))
data = data %>% mutate(sum_gly = rowSums(across(c("GGA..Gly.","GGC..Gly.", "GGG..Gly.", "GGT..Gly."))))
data = data %>% mutate(sum_leu = rowSums(across(c("TTA..Leu." , "TTG..Leu."))))
data = data %>% mutate(sum_asn = rowSums(across(c("AAC..Asn.",  "AAT..Asn."))))
data$pro_frac = data$sum_pro/data$sum
data$phe_frac = data$sum_phe/data$sum
data$leu_frac = data$sum_leu/data$sum
data$gly_frac = data$sum_gly/data$sum
data$lys_frac = data$sum_lys/data$sum
data$asn_frac = data$sum_asn/data$sum

ggboxplot(data, x = "Class", y = "pro_frac",
                color = "Class", palette=c("#6760db", "#7849bf", "#9145c4", "#c73a69", "#c2464c"), order=c("Actinopterygii", "Amphibia", "Reptilia", "Mammalia","Aves"),
                add = "jitter", shape = "Class")
ggboxplot(data, x = "Class", y = "phe_frac",
                color = "Class", palette=c("#6760db", "#7849bf", "#9145c4", "#c73a69", "#c2464c"), order=c("Actinopterygii", "Amphibia", "Reptilia", "Mammalia","Aves"),
                add = "jitter", shape = "Class")
ggboxplot(data, x = "Class", y = "leu_frac",
                color = "Class", palette=c("#6760db", "#7849bf", "#9145c4", "#c73a69", "#c2464c"), order=c("Actinopterygii", "Amphibia", "Reptilia", "Mammalia","Aves"),
                add = "jitter", shape = "Class")
ggboxplot(data, x = "Class", y = "gly_frac",
                color = "Class", palette=c("#6760db", "#7849bf", "#9145c4", "#c73a69", "#c2464c"), order=c("Actinopterygii", "Amphibia", "Reptilia", "Mammalia","Aves"),
                add = "jitter", shape = "Class")
ggboxplot(data, x = "Class", y = "lys_frac",
                color = "Class", palette=c("#6760db", "#7849bf", "#9145c4", "#c73a69", "#c2464c"), order=c("Actinopterygii", "Amphibia", "Reptilia", "Mammalia","Aves"),
                add = "jitter", shape = "Class")
ggboxplot(data, x = "Class", y = "asn_frac",
                color = "Class", palette=c("#6760db", "#7849bf", "#9145c4", "#c73a69", "#c2464c"), order=c("Actinopterygii", "Amphibia", "Reptilia", "Mammalia","Aves"),
                add = "jitter", shape = "Class")
```

# Exploring Actinopterygii
```{r}
tm = read.table('../Data/1raw/FishBaseTemperature.txt', header=T)
tm = tm %>% group_by(Species) %>% summarise(temperature = median(Temperature))
temp=merge(sinnuccontentn, tm)
cor.test(temp$temperature, temp$meanTGACskew, method = "spearman")

f=ggscatter(temp, x = "temperature", y = "meanTGACskew",
          color = "#eb4034", # Points color, shape and size
          add = "reg.line",  # Add regressin line
          add.params = list(color = "black", fill = "lightgray"), # Customize reg. line
          conf.int = TRUE, # Add confidence interval
          xlab="Mean annual water temperature, ?C")
f

```